fhir-persistence-jdbc/src/main/java/com/ibm/fhir/persistence/jdbc/util/JDBCQueryBuilder.java

```diff
@@ -1194,12 +1194,18 @@ private SqlQueryData processTokenParm(Class<?> resourceType, QueryParameter quer
         StringBuilder whereClauseSegment = new StringBuilder();
         String operator = this.getOperator(queryParm, EQ);
         boolean parmValueProcessed = false;
+        boolean appendEscape;
         SqlQueryData queryData;
         List<Object> bindVariables = new ArrayList<>();
         String tableAlias = paramTableAlias;
+        String queryParmCode = queryParm.getCode();
 
-        String code = queryParm.getCode();
-        if (!QuerySegmentAggregator.ID.equals(code)) {
+        if (!QuerySegmentAggregator.ID.equals(queryParmCode)) {
+
+            // Append the suffix for :text modifier
+            if (Modifier.TEXT.equals(queryParm.getModifier())) {
+                queryParmCode += SearchConstants.TEXT_MODIFIER_SUFFIX;
+            }
```

Comment:
I double-check and this is consistent with how we handled the `:identifier` modifier for reference search